syntax = "proto3";

option csharp_namespace = "GrpcService";

package greet;

// The greeting service definition.
service Greeter {
  // Sends a greeting
    rpc EchoBidir (stream EchoRequest) returns (stream EchoReply);
    rpc EchoBidirHangup (stream EchoRequest) returns (stream EchoReply);
    rpc EchoBidirAutoHacks (stream EchoRequest) returns (stream EchoReply);
  rpc EchoUnary (EchoRequest) returns (EchoReply);
}

// The request message containing the user's name.
message EchoRequest {
  string request = 1;
}

// The response message containing the greetings.
message EchoReply {
  string reply = 1;
}
